tests/test_generator.py

```diff
@@ -0,0 +1,361 @@
+        / f"{app.name}-{1}"
+        / "log"
+    )
+    assert osp.isdir(expected_run_path)
+    assert osp.isdir(expected_log_path)
+    assert osp.isfile(osp.join(expected_log_path, "smartsim_params.txt"))
+
+
+def test_exp_private_generate_method_app(wlmutils, test_dir, generator_instance):
+    """Test that Job directory was created from Experiment."""
+    exp = Experiment(name="experiment_name", exp_path=test_dir)
+    app = Application("name", "python", "RunSettings")
+    launch_settings = LaunchSettings(wlmutils.get_test_launcher())
+    job = Job(app, launch_settings)
+    job_execution_path = exp._generate(generator_instance, job, 1)
+    assert osp.isdir(job_execution_path)
+    head, _ = os.path.split(job_execution_path)
+    expected_log_path = pathlib.Path(head) / "log"
+    assert osp.isdir(expected_log_path)
+
+
+def test_generate_copy_file(fileutils, wlmutils, test_dir):
+    # Create the Job and attach generator file
+    launch_settings = LaunchSettings(wlmutils.get_test_launcher())
+    app = Application("name", "python", "RunSettings")
+    script = fileutils.get_test_conf_path("sleep.py")
+    app.attach_generator_files(to_copy=script)
+    job = Job(app, launch_settings)
+
+    # Create the experiment
+    experiment_path = osp.join(test_dir, "experiment_name")
+    gen = Generator(exp_path=experiment_path, run_id="temp_run")
+    path = gen.generate_job(job, 1)
+    expected_file = pathlib.Path(path) / "sleep.py"
+    assert osp.isfile(expected_file)
+
+
+def test_generate_copy_directory(wlmutils, test_dir, get_gen_copy_file):
+    # Create the Job and attach generator file
+    launch_settings = LaunchSettings(wlmutils.get_test_launcher())
+    app = Application("name", "python", "RunSettings")
+    app.attach_generator_files(to_copy=get_gen_copy_file)
+    job = Job(app, launch_settings)
+
+    # Create the experiment
+    experiment_path = osp.join(test_dir, "experiment_name")
+    gen = Generator(exp_path=experiment_path, run_id="temp_run")
+    path = gen.generate_job(job, 1)
+    expected_file = pathlib.Path(path) / "mock.txt"
+    assert osp.isfile(expected_file)
+
+
+def test_generate_symlink_directory(wlmutils, test_dir, get_gen_symlink_file):
+    # Create the Job and attach generator file
+    launch_settings = LaunchSettings(wlmutils.get_test_launcher())
+    app = Application("name", "python", "RunSettings")
+    # Path of directory to symlink
+    symlink_dir = get_gen_symlink_file
+    # Attach directory to Application
+    app.attach_generator_files(to_symlink=symlink_dir)
+    # Create Job
+    job = Job(app, launch_settings)
+
+    # Create the experiment
+    experiment_path = osp.join(test_dir, "experiment_name")
+    gen = Generator(exp_path=experiment_path, run_id="temp_run")
+    # Generate Experiment file structure
+    job_path = gen.generate_job(job, 1)
+    expected_folder = pathlib.Path(job_path) / "to_symlink_dir"
+    assert osp.isdir(expected_folder)
```

Comment:
If you want, I'd add a few more sanity checks here, just to make sure the expected dir is a symlink and is pointing at the directory we expect:
```suggestion
    expected_folder = pathlib.Path(job_path) / "to_symlink_dir"
    assert osp.isdir(expected_folder)
    assert expected_folder.is_symlink()
    assert os.fspath(expected_folder.resolve()) == osp.realpath(symlink_dir)
```